tidy3d/components/geometry/polyslab.py

```diff
@@ -11,12 +11,18 @@
 from matplotlib import path
 
 from ..base import cached_property
-from ..types import Axis, Bound, PlanePosition, ArrayFloat2D
+from ..types import Axis, Bound, PlanePosition, ArrayFloat2D, Coordinate, MatrixReal4x4, Shapely
 from ...log import log
 from ...exceptions import SetupError, ValidationError
 from ...constants import MICROMETER, fp_eps
 
 from . import base
+from . import triangulation
+
+try:
```

Comment:
is there a way to unify these attempted trimesh imports? they seem to pop up in many different files, maybe just put them in a single function if it makes sense to?

Reply:
or try to import trimesh in one module, which sets the `trimesh_available` boolean. if `trimesh` is not found, it also defines a global variable `trimesh=None`, and then other modules import trimesh from that module?

Reply:
I've created a trimesh in types that can be used elsewhere, otherwise, because of circular imports, we'd have to use it as `base.trimesh` or similar. As a bonus, we get to define `TrimeshType` in the correct place now.

Reply:
great!